src/query/allowed_objects.c

```diff
@@ -224,9 +248,38 @@ int primary_arg_index(Oid funcoid)
   FAILWITH("Cannot identify the primary argument position for funcid %u.", funcoid);
 }
 
-bool is_allowed_cast(Oid funcoid)
+bool is_allowed_cast(const FuncExpr *func_expr)
 {
-  return is_funcname_member_of(funcoid, g_allowed_casts, ARRAY_LENGTH(g_allowed_casts));
+  if (is_funcname_member_of(func_expr->funcid, g_allowed_casts, ARRAY_LENGTH(g_allowed_casts)))
+  {
+    return true;
+  }
+  else if (is_funcname_member_of(func_expr->funcid, g_decimal_integer_casts, ARRAY_LENGTH(g_decimal_integer_casts)))
+  {
+    /*
+     * Special case, where a `numeric_int4` cast is called on variants of `extract` which return
+     * integer numbers, e.g. `cast(extract(minute from ...) as integer)`.
+     */
+    Node *cast_arg = linitial(func_expr->args);
+    if (IsA(cast_arg, FuncExpr))
+    {
+      FuncExpr *cast_arg_expr = (FuncExpr *)cast_arg;
+      if (is_funcname_member_of(cast_arg_expr->funcid, g_extract_functions, ARRAY_LENGTH(g_extract_functions)) ||
+          cast_arg_expr->funcid == F_DATE_PART_TEXT_DATE)
+      {
+        Node *extract_field = linitial(cast_arg_expr->args);
+        if (IsA(extract_field, Const))
+        {
+          Const *extract_field_const = (Const *)extract_field;
+          Assert(extract_field_const->consttype == TEXTOID);
+          const char *field = TextDatumGetCString(extract_field_const->constvalue);
+
+          return is_member_of(field, g_integer_extract_fields, ARRAY_LENGTH(g_integer_extract_fields));
```

Comment:
I think we can consider all date fields as safe for casting to integer (so no need to check the operation type).

Reply:
The reason for the check is that some fields return a legitimate floating point / decimal, so the cast is actually a round. Not sure about whether it is absolutely required, but I think we try to disallow these in `pg_diffix`.

Reply:
But I guess if you're saying it still makes sense to allow, then let's allow, I was reluctant to initially, but maybe the simplicity is worth more than air-tightness

Reply:
Done, I've also spotted another place where int2/8 were treated inconsistently, so I fixed it